src/js/components/productSupplier/modals/AddAttributeOptionModal.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+
+import PropTypes from 'prop-types';
+import { RiCloseFill } from 'react-icons/all';
+import Modal from 'react-modal';
+
+import Button from 'components/form-elements/Button';
+import TextInput from 'components/form-elements/v2/TextInput';
+import useTranslate from 'hooks/useTranslate';
+
+import '../styles.scss';
+
+const AddAttributeOptionModal = ({
+  isOpen,
+  close,
+  selectedAttribute,
+  setValue,
+}) => {
+  const [inputValue, setInputValue] = useState(null);
+  const translate = useTranslate();
+
+  const onChange = (e) => setInputValue(e.target.value);
+
+  const onClose = () => {
+    setInputValue(null);
+    close();
+  };
+
+  const setAttributeValue = (attribute, value) => () => {
+    setValue(`attributes.${selectedAttribute?.id}`, { id: attribute?.id, value, label: value });
+    setInputValue(null);
+    close();
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      className="modal-content w-50"
+      shouldCloseOnOverlayClick={false}
+    >
+      <div>
+        <div className="d-flex justify-content-between mb-3">
+          <h4>{selectedAttribute?.name}</h4>
+          <RiCloseFill
+            size="24px"
+            className="cursor-pointer"
+            role="button"
+            onClick={onClose}
+          />
+        </div>
+        <div>
+          <TextInput
+            required
+            title={{ id: '', defaultMessage: 'Other' }}
+            errorMessage={inputValue === '' && translate(
```

Comment:
what if `inputValue` would be `null`? Why don't we want to validate it then?

Reply:
If the value is null, we are treating the input as not touched previously